Create guidelines for improved repo quality

dorukaks opened the pull request:
## Purpose
**_Explain the goal of this PR, if it addresses an existing issue be sure to link to it._**
Creating guidelines for improved communication quality.

**_Tag each of the issues that you are addressing with this PR with "Closes #<issueNumber>"_**
Closes #3 

**_Describe the big picture of your changes here, perhaps using a bullet list if multiple changes are done to accomplish a single goal._**
_If it accomplishes multiple goals, it would be best to create separate PR's for each._ 
- Created `./formats` directory for guidelines 
- Created `commits.md` for commit guidelines


## Type of change
**_What types of change is it?_**
_Select the appropriate type(s) that describe this PR_

- New feature (non-breaking change which adds functionality)
- Repository chores (changes made to improve repository/workflow quality)


## Checklist
**_Put an `x` in the boxes that apply._**

- [ ] My code functions as intended
- [ ] I have used camelCase in my variable and function names
- [ ] I have commented my code properly
- [ ] I have provided the necessary prompts for automatic documentation
- [ ] I have provided a toy case under the examples folder
- [x] I have addressed all the issues entirely

dorukaks committed:
docs(commits): create commit template document

Created a guideline that will be used as a reference
while writing commit mesages.

dorukaks committed:
docs(commits): create commit template document

Created a guideline that will be used as a reference
while writing commit mesages.

dorukaks committed:
docs(commits): fix formatting
Markdown didn't look as intended, sending a quick fix.

dorukaks committed:
docs(commits): fix formatting
Markdown didn't look as intended, sending a quick fix.

dorukaks committed:
docs(commits): add negative examples
Included negative examples for educational purposes.

dorukaks committed:
docs(commits): add negative examples
Included negative examples for educational purposes.

dorukaks committed:
chore(templates): merge templates from main

Merging remote-tracking branch 'origin' into chore/doruk/3

dorukaks committed:
(chore) create branching guideline

Created a markdown document to create the initial version of
branching guidelines.

dorukaks committed:
(chore) create branching guideline

Created a markdown document to create the initial version of
branching guidelines.

sahilbhola14 commented on formats/commits.md:
```diff
@@ -0,0 +1,54 @@
+
+_subject_: Brief summary(30-40 characters) of the changes with couple words. Use imperative mood. Don't end with a period.
+
+## Tags for commit messages
```
I think using the following commit message style has more readability,
(Type of commit/ scope)

dorukaks replied:
changed the commit message structure to "(tag) subject" in [0eaa6cd](https://github.com/PinguDevTeam/ICF-MI-framework/pull/5/commits/0eaa6cd85c91afe9c52c70029b2a88792bf2bfa7)

dorukaks committed:
(docs) merge remote-tracking branch 'origin/main' into chore/doruk/3

dorukaks commented on formats/commits.md:
```diff
@@ -0,0 +1,54 @@
+
+_subject_: Brief summary(30-40 characters) of the changes with couple words. Use imperative mood. Don't end with a period.
+
+## Tags for commit messages
```
We might need to add another tag for merging but not sure

sahilbhola14 replied:
I think 'CI' tag is used for that

dorukaks replied:
Added CI tag for merging from parent branch :+1:

dorukaks replied:
made appropriate changes in examples in [fc41d9c](https://github.com/PinguDevTeam/ICF-MI-framework/pull/5/commits/fc41d9c646e99e8a23370d65e2d446ae2dc1f05e)

dorukaks committed:
(fix) correct merge command
Changed the example merge command to merge with the remote repository

dorukaks committed:
(fix) correct merge command
Changed the example merge command to merge with the remote repository

dorukaks commented:
@sahilbhola14 have a look at this as soon as you have time and quickly discuss the structure before we start to branch out for our tasks. I have an alternate repository structure in mind so maybe we can decide between those two. Shouldn't take much time (~10 mins)

sahilbhola14 commented on formats/branching.md:
````diff
@@ -0,0 +1,59 @@
+
+The naming convention for each branch will have the following structure 
+
+```<tag>/<your name>/<number of issue(s) you are addressing>```
````
I think writing # issue id will be more useful, thoughts?

dorukaks replied:
You are right, it's ambiguous this way :+1:

dorukaks replied:
addressed through 82c7b95

sahilbhola14 commented on formats/branching.md:
```diff
@@ -0,0 +1,59 @@
+
+## General Workflow
+
+Each time you start to work on a branch, please `fetch` all the changes from the reposit using `git fetch` and check if there are any changes affecting your current active branch using `git status`.
```
I think it would be more useful to add the following as well,
Git log, Git add, Git commit, Git status, Git push, Git branch -d <name of branch>

dorukaks replied:
and also `git branch`

sahilbhola14 replied:
Added git log description in [346aaa2](https://github.com/PinguDevTeam/ICF-MI-framework/pull/5/commits/346aaa2d131774fdc6fe814daec1b9f66f2cd327)

dorukaks replied:
Addressed through [58627e7](https://github.com/PinguDevTeam/ICF-MI-framework/pull/5/commits/58627e7b331905d7ff9e759ed5fe2b1b6862a99d)...[634ed7e](https://github.com/PinguDevTeam/ICF-MI-framework/pull/5/commits/634ed7ea924a08ad9bef3c7d903e85d55352492b)

sahilbhola14 committed:
(docs) Add git log description

sahilbhola14 committed:
(docs) Add git log description

dorukaks commented on formats/branching.md:
```diff
@@ -0,0 +1,61 @@
+# Branching and Pull Requests
+
+This markdown file describes the structure we will follow for branches and how we will tackle the pull requests.
```
@sahilbhola14 please add any necessary subdirectories that I forgot mentioning here

dorukaks committed:
(chore) create workflow guideline

Created a separate workflow.md file for workflow related guidelines

dorukaks committed:
(chore) create workflow guideline

Created a separate workflow.md file for workflow related guidelines

sahilbhola14 committed:
(docs/README) Add auto-docum. documentation

Added how to run sphinx (auto-documentation) tool.

dorukaks committed:
(fix) remove workflow related parts

Removed workflow related parts of the guideline to workflow.md

dorukaks committed:
(fix) remove workflow related parts

Removed workflow related parts of the guideline to workflow.md

sahilbhola14 committed:
(build!) Delete TT-ICE

TT-ICE was deleted as the hyphen created documentation issues

dorukaks committed:
(fix) change repository structure

Modified repository structure related parts accordingly

dorukaks committed:
(fix) change repository structure

Modified repository structure related parts accordingly

sahilbhola14 committed:
(feat) Add __init__.py files for modules

Added __init__.py files to each sub-directory such that
the auto-doc tool will read the module

dorukaks committed:
(fix) modify issue guideline

Fixed issue creation and branching related parts of the guideline

dorukaks committed:
(fix) modify issue guideline

Fixed issue creation and branching related parts of the guideline

sahilbhola14 committed:
(feat/doc_templates) Create doc templates

Created doc templates for each modele viz. GIM, LSTM, TT_ICE

sahilbhola14 committed:
(feat/scripts) Create automation scripts for doc.

Created automation scripts for documenting the code

dorukaks committed:
(chore) add continuous integration (CI) tag

Added CI taf to commit tags

dorukaks committed:
(chore) add continuous integration (CI) tag

Added CI taf to commit tags

dorukaks committed:
(fix) change commit message format

Changed commit message format  and applied changes throughout document

dorukaks committed:
(fix) change commit message format

Changed commit message format  and applied changes throughout document

dorukaks committed:
(fix) change format of examples

Changed the format of negative example to match rest of the guidelines

dorukaks committed:
(fix) change format of examples

Changed the format of negative example to match rest of the guidelines

sahilbhola14 committed:
(fix/create_documentation) Recompile doc string

Made sure the doc strings are processed each time the code
is compiled.
index.rst and introduction.rst are preserved

dorukaks committed:
(chore) add issue creation

Added "creating an issue" to the guideline

dorukaks committed:
(chore) add issue creation

Added "creating an issue" to the guideline

dorukaks committed:
(chore) add branch creation

Added "creating a branch" to workflow guideline

dorukaks committed:
(chore) add branch creation

Added "creating a branch" to workflow guideline

dorukaks committed:
(chore) add git fetch

Added guideline for "git fetch"

dorukaks committed:
(chore) add git fetch

Added guideline for "git fetch"

dorukaks committed:
(chore) add git status

Added guidelines for "git status"

dorukaks committed:
(chore) add git status

Added guidelines for "git status"

dorukaks committed:
(chore) add staging guidelines

Added detailes on staging changes

dorukaks committed:
(chore) add staging guidelines

Added detailes on staging changes

dorukaks committed:
(chore) add git commit guideline

Added information on committing staged changes

dorukaks committed:
(chore) add git commit guideline

Added information on committing staged changes

dorukaks committed:
(chore) add reverting commits

Added information on how to revert an accidental commit

dorukaks committed:
(chore) add reverting commits

Added information on how to revert an accidental commit

dorukaks committed:
(chore) add git push

Added information on how to push commited changes to remote

dorukaks committed:
(chore) add git push

Added information on how to push commited changes to remote

dorukaks committed:
(chore) add git log information

Added information on how to display the history of a branch

dorukaks committed:
(chore) add git log information

Added information on how to display the history of a branch

dorukaks reviewed (commented):
This PR is ready to review.

sahilbhola14 committed:
(style) Rename TT_ICE to TTICE

sahilbhola14 reviewed (approved):
Looks good to me!

sahilbhola14 committed:
Merge pull request #18 from PinguDevTeam/feat/sahil/6

(feat/automatic doc) Automatic documentation

Automatic documentation was added in this feature
Closes #6
Branch name: Feat/sahil/6
Reviewers: Doruk, Codie

dorukaks committed:
(CI) Merge remote-tracking branch 'refs/remotes/origin/chore/doruk/3' into chore/doruk/3

dorukaks commented:
Rebased the branch to current main branch per @sahilbhola14 's request (happened offline). Merging the PR.
